cpp/daal/src/algorithms/dtrees/forest/df_train_dense_default_impl.i

```diff
@@ -1084,22 +1084,63 @@ bool TrainBatchTaskBase<algorithmFPType, BinIndexType, DataHelper, cpu>::findBes
                                                                                              typename DataHelper::TSplitData & bestSplit,
                                                                                              algorithmFPType totalWeights)
 {
-    chooseFeatures();
-    size_t nVisitedFeature       = 0;
-    const size_t maxFeatures     = nFeatures();
-    const float qMax             = 0.02; //min fracture of observations to be handled as indexed feature values
-    IndexType * bestSplitIdx     = featureIndexBuf(0) + iStart;
-    IndexType * aIdx             = _aSample.get() + iStart;
-    int iBestSplit               = -1;
-    int idxFeatureValueBestSplit = -1; //when sorted feature is used
+    /* counter of the number of visited features, we visit _nFeaturesPerNode
+    *  depending on _par.useConstFeatures constant features can be skipped
+    */
+    size_t nVisitedFeature = 0;
+    /* total number of features */
+    const size_t maxFeatures = nFeatures();
+    /* (??) min fracture of observations to be handled as indexed feature values */
+    const float qMax = 0.02;
```

Comment:
This number is looking weird t me. I see that it have been in place for a quite some time already but it still looks magic. Should we consider changing this thing? (Not in this PR though)

Reply:
And another question is why it is `float` not `algorithmFPType`?

Reply:
All good points. See also Ian's comments above. I'm happy to change this with this PR. But I don't have any opinion on "how" to change it.